File: frontend/src/services/dict.ts
import api from './api';

export interface Dict {
  id: string;
  name: string;
  code: string;
  description: string;
  status: number;
  created_at: string;
  updated_at: string;
}

export interface DictItem {
  id: string;
  dict_id: string;
  label: string;
  value: string;
  sort: number;
  status: number;
  created_at: string;
  updated_at: string;
}

export interface CreateDictRequest {
  name: string;
  code: string;
  description?: string;
  status?: number;
}

export interface UpdateDictRequest {
  name?: string;
  code?: string;
  description?: string;
  status?: number;
}

export interface CreateDictItemRequest {
  dict_id: string;
  label: string;
  value: string;
  sort?: number;
  status?: number;
}

export interface UpdateDictItemRequest {
  dict_id?: string;
  label?: string;
  value?: string;
  sort?: number;
  status?: number;
}

export interface DictWithItems extends Dict {
  items: DictItem[];
}

export const dictService = {
  // 字典管理
  getDicts: async (): Promise<Dict[]> => {
    const response = await api.get('/dicts');
    return response.data.dicts;
  },

  getDict: async (id: string): Promise<Dict> => {
    const response = await api.get(`/dicts/${id}`);
    return response.data;
  },

  createDict: async (data: CreateDictRequest): Promise<Dict> => {
    const response = await api.post('/dicts', data);
    return response.data.dict;
  },

  updateDict: async (id: string, data: UpdateDictRequest): Promise<Dict> => {
    const response = await api.put(`/dicts/${id}`, data);
    return response.data.dict;
  },

  deleteDict: async (id: string): Promise<void> => {
    await api.delete(`/dicts/${id}`);
  },

  // 字典项管理
  getDictItems: async (dictId: string): Promise<DictItem[]> => {
    const response = await api.get(`/dicts/${dictId}/items`);
    return response.data.items;
  },

  getDictItem: async (id: string): Promise<DictItem> => {
    const response = await api.get(`/dict-items/${id}`);
    return response.data;
  },

  createDictItem: async (data: CreateDictItemRequest): Promise<DictItem> => {
    const response = await api.post('/dict-items', data);
    return response.data.item;
  },

  updateDictItem: async (id: string, data: UpdateDictItemRequest): Promise<DictItem> => {
    const response = await api.put(`/dict-items/${id}`, data);
    return response.data.item;
  },

  deleteDictItem: async (id: string): Promise<void> => {
    await api.delete(`/dict-items/${id}`);
  },

  // 一次性获取所有字典和字典项
  getAllDictsWithItems: async (): Promise<DictWithItems[]> => {
    const response = await api.get('/dicts/all-with-items');
    return response.data.dicts;
  },
};

File: backend/internal/model/dict.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type Dict struct {
	ID          uint           `json:"id" gorm:"primaryKey"`
	Name        string         `json:"name" gorm:"not null" binding:"required"`
	Code        string         `json:"code" gorm:"uniqueIndex;not null" binding:"required"`
	Description string         `json:"description"`
	Status      int            `json:"status" gorm:"default:1"` // 1:正常 0:禁用
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `json:"-" gorm:"index"`

	// 关联关系
	Items []DictItem `json:"items" gorm:"foreignKey:DictID"`
}

type DictItem struct {
	ID        uint           `json:"id" gorm:"primaryKey"`
	DictID    uint           `json:"dict_id" gorm:"not null"`
	Label     string         `json:"label" gorm:"not null" binding:"required"`
	Value     string         `json:"value" gorm:"not null" binding:"required"`
	Sort      int            `json:"sort" gorm:"default:0"`
	Status    int            `json:"status" gorm:"default:1"` // 1:正常 0:禁用
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `json:"-" gorm:"index"`

	// 关联关系
	Dict Dict `json:"dict" gorm:"foreignKey:DictID"`
}
